gax/src/main/java/com/google/api/gax/batching/BatcherImpl.java

```diff
@@ -100,7 +100,10 @@ public BatcherImpl(
     Preconditions.checkNotNull(executor, "executor cannot be null");
     currentOpenBatch = new Batch<>(prototype, batchingDescriptor, batchingSettings);
 
-    long delay = batchingSettings.getDelayThreshold().toMillis();
+    long delay =
+        batchingSettings.getDelayThreshold() == null
+            ? 1L
+            : batchingSettings.getDelayThreshold().toMillis();
```

Comment:
Flushing every 1 ms since pretty excessive. Where did this come from?

Reply:
I went with the existing [BatchingSettings](https://github.com/googleapis/gax-java/blob/master/gax/src/main/java/com/google/api/gax/batching/BatchingSettings.java#L117) default value.

We can set this to `100ms`, Anyways this will be overridden by client-specific defaults value.

Reply:
Oh, I missed that. I think that we need to follow existing semantics. The default in BatchingSettings is 1ms. So if it's null here then we should disable autoflush

Reply:
I think it could be as simple as:

Long delay = batchingSettings.getDelayThreshold();
if (delay != null) {
  scheduledFuture =
        executor.scheduleWithFixedDelay(runnable, delay, delay, TimeUnit.MILLISECONDS);	        executor.scheduleWithFixedDelay(runnable, delay, delay, TimeUnit.MILLISECONDS);
} else {
  scheduledFuture = Futures.immediateCanceledFuture()
}

Reply:
Agreed. have incorporated this.